utils/mgmt_system.py

```diff
@@ -1144,6 +1145,27 @@ def deploy_template(self, template, *args, **kwargs):
     def remove_host_from_cluster(self, hostname):
         raise NotImplementedError('remove_host_from_cluster not implemented')
 
+    def mark_as_template(self, vm_name):
```

Comment:
The innocent-sounding name doesn't really line up with "Take a VM, stop it, turn it into a template, and delete it." This should be solved with a docstring explaining what this does and why it's needed.